temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInfoImpl.java

```diff
@@ -19,15 +19,40 @@
 
 package io.temporal.internal.sync;
 
+import io.temporal.api.common.v1.Payload;
 import io.temporal.api.common.v1.SearchAttributes;
 import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.common.converter.DefaultDataConverter;
 import io.temporal.internal.replay.ReplayWorkflowContext;
 import io.temporal.workflow.WorkflowInfo;
 import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
 final class WorkflowInfoImpl implements WorkflowInfo {
 
+  public enum SearchAttribute {
+    ExecutionStatus,
+    CloseTime,
+    CustomBoolField,
+    CustomDatetimeField,
+    CustomNamespace,
+    CustomDoubleField,
+    CustomIntField,
+    CustomKeywordField,
+    CustomStringField,
+    NamespaceId,
+    ExecutionTime,
+    HistoryLength,
+    RunId,
+    StartTime,
+    TaskQueue,
+    WorkflowId,
+    WorkflowType;
+  }
```

Comment:
I don't think we can hardcode the list of the search attributes as they can be added to a cluster at any time.